src/passes/OptimizeInstructions.cpp

```diff
@@ -261,6 +261,20 @@ struct OptimizeInstructions
     inReplaceCurrent = false;
   }
 
+  // The override of replaceCurrent() above us ensures that we revisit a node
+  // after we optimize it, allowing us to apply multiple rules while we still
+  // find things to do. However, we are still doing a post-walk, which means we
+  // do not backtrack. Sometimes a pattern creates several nested children, in
+  // which case it would make sense to re-optimize them instead of leaving
+  // things for a later run of this pass, which this function does. This
+  // potentially makes the pass not linear time, so it should be used sparingly.
+  void reOptimize(Expression*& curr) {
```

Comment:
Where is this used?

Reply:
It is no longer used, but it will be useful in the future I think. If you object I can move it to another PR, but it causes no harm here I believe (no risk of affecting bisection or anything, which is the main reason I prefer small PRs).

Reply:
It seems better to bring in the code with a PR that can use and test it, but beyond that I agree there is no harm.